src/main/java/com/books/services/BookService.java

```diff
@@ -60,17 +72,34 @@ public void addBook(Book book) {
         repository.add(book);
     }
 
-    public void setBook(int id, Book book) {
-        repository.setItem(id, book);
+    public void setBook(Book book) {
+        repository.setItem(book.getId(), book);
     }
+
     public void removeAuthorBook(int bookId, int authorId) {
         Book book = getBookById(bookId);
         book.removeAuthor(authorId);
+        setBook(book);
+    }
+
+    public void changePublisher(int bookId, int publisherId) {
+        Book book = getBookById(bookId);
+        book.setPublisher(publisherService.getPublisherById(publisherId));
+        setBook(book);
+    }
+
+    public void removeBook(int bookId) {
+        repository.remove(bookId);
     }
+
     public void addAuthorBook(int bookId, int authorId) {
         Book book = getBookById(bookId);
-        AuthorService authorService = new AuthorService();
+
         Person author = authorService.getAuthorById(authorId);
-        book.getAuthors().add(author);
+        //TODO why it doesn't add? (doesn't save in DB)
+        // book.getAuthors().add(author);
```

Comment:
If you still have questions, you could ask me directly. If this is trash comment, please remove it